### ISSUE TITLE
Changing sample rate after initialisation

### ISSUE THREAD

**Issue by sfjohnson** [avaneev/r8brain-free-src#17]
Hi,

Thanks for your years of work on this excellent library. I'm currently using it for my network audio project as a resampling stage before and after Opus encode/decode. I'd also like to use it to correct clock drift between the sender and receiver.

Is there a function to change the sample rate ratio after the `CDSPResampler` has had samples go through it? I don't really need a continuous change. I'll be monitoring the receive buffer and adjusting the ratio when necessary to keep it from underflowing or overflowing.

I did something like this with your library about 6 years ago. I don't think that function exists any more though and it's been so long I can't find my old code.

Thanks,
Sam.

**Comment by avaneev** [comment]
This is currently impossible. In the past a function was available, but it was an incorrect way to adjust sample rate, because it did not adjust filters accordingly. A working approach is to create a second resampler the moment sample rate change is necessary, then feed both resamplers with the same data and do a cross-fade, then switch to the second resampler completely and delete the previous resampler. You can use a value returned from the getInLenBeforeOutStart() function for synchronization.

**Comment by sfjohnson** [comment]
Thanks for the information, I'll try this approach.

**Comment by sfjohnson** [comment]
@avaneev I'm finally getting around to implementing your solution above. I noticed that op0 here is not const https://github.com/avaneev/r8brain-free-src/blob/master/CDSPResampler.h#L555

Am I allowed to modify this buffer (without overflowing its length) without corrupting the internal state of the resampler? I'm mixing the output of two resamplers together, and it would be good to avoid allocating an extra destination buffer.

Thanks.
